### PR TITLE
bring annotations to API

### PR DESCRIPTION
**Pull request by tcitworld**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | WIP
| Documentation | no
| Translation   | no
| Fixed tickets | https://github.com/wallabag/wallabag/issues/1800
| License       | MIT

### CODE REVIEW

File: src/Wallabag/ApiBundle/Controller/WallabagRestController.php [hunk 1]
```diff
@@ -344,6 +346,138 @@ public function deleteTagAction(Tag $tag)
...
+     *      requirements={
+     *          {"name"="entry", "dataType"="integer", "requirement"="\w+", "description"="The entry ID"}
+     *      }
+     * )
+     *
+     * @return Response
+     */
+    public function getAnnotationsAction(Entry $entry)
+    {
+
```
Comment (j0k3r) [inline src/Wallabag/ApiBundle/Controller/WallabagRestController.php]: You have several extra unnecessary lines like that in this file

**Comment by tcitworld** [comment]
Can I call from APIBundle/RestController to AnnotationBundle/Controller instead of having twice the same functions ?

**Comment by nicosomb** [comment]
Which one? 
Maybe you can put this function into a helper class and inject it via a service?

**Comment by j0k3r** [comment]
If we think more globally.
The AnnontationBundle is just an API for annotator.js, right?
Why don't we move that stuff in ApiBundle with Entity & Repository in CoreBundle?

**Comment by tcitworld** [comment]
Well, thing is Annotator.js doesn't go with OAuth, So we need separate routes just for the annotations made/retrieved by the interface.

**Comment by nicosomb** [comment]
@tcitworld need help for tests?

**Comment by tcitworld** [comment]
It's just as @j0k3r said, there must be a more efficient way to do this without having twice the same function. On one side we have an API (without auth atm, but compatible with [this](https://github.com/heckj/authz) it seems) just for annotations in the web interface and on the other one there's our global API that requires O'Auth.

**Comment by nicosomb** [comment]
(PR replaced by https://github.com/wallabag/wallabag/pull/1890)

**Pull request closed without merging**